File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/core/AdditionalFiles.java
package app.zxtune.core;

import androidx.annotation.Nullable;

import java.nio.ByteBuffer;

/**
 * Interface to additional files access
 */
public interface AdditionalFiles {
  /**
   * @return null if no additional files required, empty array if all the files are resolved,
   * else list of required filenames relative to current one
   */
  @Nullable
  String[] getAdditionalFiles();

  /**
   * @param name additional file name as returned by ListAdditionalFiles
   * @param data additional file content
   */
  void resolveAdditionalFile(String name, ByteBuffer data);
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/core/AdditionalFiles.kt
package app.zxtune.core

import java.nio.ByteBuffer

/**
 * Interface to additional files access
 */
interface AdditionalFiles {
    /**
     * @return null if no additional files required, empty array if all the files are resolved,
     * else list of required filenames relative to current one
     */
    val additionalFiles: Array<String>?

    /**
     * @param name additional file name as returned by additionalFiles
     * @param data additional file content
     */
    fun resolveAdditionalFile(name: String, data: ByteBuffer)
}
